roles/repository/templates/share-config-custom.xml

```diff
@@ -76,13 +76,15 @@
             does not rewrite the Referer header.
          -->
 
-         <referer></referer>
+	 {# Compute referers & origins -#}
+         {% import 'csrf_macros.j2' as _csrf %}
+	 <referer>{{ _csrf.compute('referer', csrf.urls)}}</referer>
 
          <!--
             Override and set this property with a regexp that if you have placed Share behind a proxy that
             does not rewrite the Origin header.
          -->
-         <origin></origin>
+	 <origin>{{ _csrf.compute('origin', csrf.urls)}}</origin>
       </properties>
 
       <!--
```

Comment:
can you rename the file to .j2 so editors won't bother with syntax highlighting?